mk/config.mk

```diff
@@ -250,10 +250,20 @@ CFG_CORE_SANITIZE_UNDEFINED ?= n
 CFG_CORE_SANITIZE_KADDRESS ?= n
 
 # Device Tree support
-# When enabled, the TEE _start function expects to find the address of a
-# Device Tree Blob (DTB) in register r2. The DT parsing code relies on
-# libfdt.  Currently only used to add the optee node and a reserved-memory
-# node for shared memory.
+#
+# When CFG_DT is enabled core embeds the FDT library (libfdt) allowing DTB
+# image parsing from the core.
+#
+# When CFG_EMBEDDED_DTS is defined, it is expected to provide the base name
+# of the device tree source file for which a device tree blob is generated
+# when core is built.
+#
+# Note: When CFG_DT is enabled, the TEE _start function expects to find
+# the address of a Device Tree Blob (DTB) in register X2/R2 provided by
+# the early boot stage or value 0 if boot stage provides no DTB.
+ifneq ($(CFG_EMBEDDED_DTS),)
```

Comment:
This is a bit confusing because it's a DTB that is embedded. From a source code point of view, in for instance `core/arch/arm/kernel/generic_boot.c`, it doesn't make sense.

Reply:
I see you point even if and don't find it confusing.

There are 2 info required: whether embedded DTB is enabled or disabled and which DTS to use if enabled. This P-R use a single directive for both.

Would you prefer `CFG_EMBED_DTB=y|n` + `CFG_EMBED_DTB_SOURCE_FILE=dts-file-name.dts`

Reply:
Yes, I'd prefer that.